src/test/java/com/crud/tasks/controller/TaskControllerTest.java

```diff
@@ -0,0 +1,127 @@
+package com.crud.tasks.controller;
+
+import com.crud.tasks.domain.Task;
+import com.crud.tasks.dto.TaskDto;
+import com.crud.tasks.mapper.TaskMapper;
+import com.crud.tasks.service.DbService;
+import com.google.gson.Gson;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.web.SpringJUnitWebConfig;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+@SpringJUnitWebConfig
+@WebMvcTest(TaskController.class)
+public class TaskControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private DbService dbService;
+
+    @MockBean
+    private TaskMapper taskMapper;
+
+    @Test
+    public void testGetTasks() throws Exception {
+        //Given
+        List<TaskDto> taskDtoList = List.of(new TaskDto(1L, "Test_title", "Content_test"));
+        when(taskMapper.mapToTaskDtoList(anyList())).thenReturn(taskDtoList);
+        //When & Then
+        mockMvc
+                .perform(MockMvcRequestBuilders
+                        .get("/v1/task/getTasks")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(1)))
```

Comment:
podpowiedź: użycie statycznych importów do metod `hasSize`, `is` itp. sprawiłoby, że łatwiej czytało by się test